hknweb/tutoring/models.py

```diff
@@ -56,9 +70,72 @@ def start_time(self):
     def end_time(self):
         return self.time(self.hour + 1)
 
+    def adjacent_to(self, other_slot):
+        return other_slot.day == self.day and abs((other_slot.hour - self.hour)) == 1
 
-class Tutor(models.Model):
+
+class CoursePreference(models.Model):
+    current = 1
+    completed = 2
+    preferred = 3
+    COURSE_PERFS = [
+        (current, 'Current'),
+        (completed, 'Completed'),
+        (preferred, 'Preferred')
+    ]
     id = models.AutoField(primary_key=True)
-    name = models.CharField(max_length=255)
-    slots = models.ManyToManyField(Slot)
-    courses = models.ManyToManyField(Course)
+    courses = models.ForeignKey(Course, on_delete=DO_NOTHING)
```

Comment:
ditto. If the course gets deleted, we probably shouldn't have preferences to a non-existent course.